core/embed/rust/src/ui/model_tr/component/input_methods/choice.rs

```diff
@@ -381,32 +472,67 @@ where
     }
 
     fn event(&mut self, ctx: &mut EventCtx, event: Event) -> Option<Self::Msg> {
+        // Possible animation movement when setting (randomizing) the page counter.
```

Comment:
Nit: the function is getting quite long, consider splitting it.

Reply:
I was thinking about this, but I could not find a simple way of doing it. The "issue" is that there are a lot of conditional returns, sometimes quite deep in the logic. So that the `event` itself can still return from any of these cases, it would need to check the return value of all of the smaller functions - and I think it would be a lot of boilerplate and moving `event` and boolean arguments around.

Even though it is quite long, it can be IMO read nicely from top to bottom to get the understanding of what is happening and what may be the edge-cases.

If you have some ideas of how to refactor it, of course, feel free to suggest anything concrete